### PR TITLE
LRP-4663 Modify the usage of ignore_zip_file

### PR DESCRIPTION
**Pull request by Wykenz**

### CODE REVIEW

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 1]
```diff
@@ -32,19 +37,39 @@ function check_if_tag_exists {
...
 }
 
+function check_ignore_argument {
+	if [[ "x${IGNORE_ZIP_FILES}" =~ x*${hotfix_zip_file}* ]]
+	then
+		lc_log DEBUG "Ignoring '${file_url}'."
+
+		return "${LIFERAY_COMMON_EXIT_CODE_OK}"
+	else
+		lc_log DEBUG "The file on '${file_url}' is not on the ignore list."
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: $fIle_url is an invalid variable in this function.

'The file on' -> Should be something like 'The file'

'is not on the ignore list.' -> Should be something like 'is not ignored via command line argument'. Something that tells more information about the reason.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 2]
```diff
@@ -32,19 +37,39 @@ function check_if_tag_exists {
...
+}
+
 function check_ignore_zip_file {
 	local hotfix_zip_file="${1}"
-	local release_version="${2}"
 
 	local file_url="${zip_directory_url}/${hotfix_zip_file}"
 
-	if [[ "x${IGNORE_ZIP_FILES}" =~ x*${hotfix_zip_file}* ]]
+	check_ignore_argument
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: argument is missing.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 3]
```diff
@@ -32,19 +37,39 @@ function check_if_tag_exists {
...
+
+		return "${LIFERAY_COMMON_EXIT_CODE_BAD}"
+	fi
+}
+
 function check_ignore_zip_file {
 	local hotfix_zip_file="${1}"
-	local release_version="${2}"
 
 	local file_url="${zip_directory_url}/${hotfix_zip_file}"
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: This is rather a path. We should rename it to path everywhere probably? It's a question.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh (hunk 2 above)
Comment (tomposmiko) [reply narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: Also something like check_ignore_via_argument Would be more talkative.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 4]
```diff
@@ -32,19 +37,39 @@ function check_if_tag_exists {
...
 
-	if [[ "x${IGNORE_ZIP_FILES}" =~ x*${hotfix_zip_file}* ]]
+	check_ignore_argument 
+	check_ignore_txt_file "${IGNORE_ZIP_FILES_PRESISTENT_FILE}"
+	check_ignore_txt_file "${IGNORE_ZIP_FILES_EXPANDING_FILE}"
+}
+
+function check_ignore_txt_file {
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: Functions with similar purpose are better the be located next to each other.

Comment (tomposmiko) [reply narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: check_ignore_via_file would be more talkative. Though still does not sound the best.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 5]
```diff
@@ -32,19 +37,39 @@ function check_if_tag_exists {
...
+	local ignore_zip_file="${1}"
+
+	if [[ $(grep -q "${hotfix_zip_file}" "${ignore_zip_file}") ]]
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: No [[ $(command)]] is needed. Actually in this case it should not even work. It should be just `if (command)`

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 6]
```diff
@@ -305,11 +330,15 @@ function print_help {
 }
 
 function process_argument_ignore_zip_files {
+	IGNORE_ZIP_FILES_PRESISTENT_FILE="$(dirname "$(readlink /proc/$$/fd/255 2>/dev/null)")/ignore_zip_files_persistent.txt"
+	IGNORE_ZIP_FILES_EXPANDING_FILE="${LIFERAY_COMMON_DOWNLOAD_CACHE_DIR}/ignore_zip_files_expanding.txt"
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: IGNORE_ZIP_FILES_CACHE_FILE would be better name.

Comment (tomposmiko) [reply narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: And it should be contained relatively, so if there is a test and a production job on the same jenkins node, they do not conflict.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 7]
```diff
@@ -305,11 +330,15 @@ function print_help {
...
+	
+	if [ ! -e "${IGNORE_ZIP_FILES_EXPANDING_FILE}" ]
+	then
+		lc_log DEBUG "Creating ${IGNORE_ZIP_FILES_EXPANDING_FILE}"
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: Put the $VARIABLE in quotes, so if it is empty for any reason, one still can observe it.
However, we don't have a rule on this and I am not sure if it's going to be approved by the superior reviewers. It's a good practice, though.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 8]
```diff
@@ -305,11 +330,15 @@ function print_help {
...
 
-	local ignore_zip_files_persistent_file="$(dirname "$(readlink /proc/$$/fd/255 2>/dev/null)")/ignore_zip_files_persistent.txt"
-	local ignore_zip_files_persistent_list=$(tr '\n' ',' < "${ignore_zip_files_persistent_file}")
-
-	IGNORE_ZIP_FILES="${IGNORE_ZIP_FILES},${ignore_zip_files_persistent_list}"
+		touch "${IGNORE_ZIP_FILES_EXPANDING_FILE}"
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: Extra tab is there and a new line is missing above the command.

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh (hunk 1 above)
Comment (Wykenz) [reply narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: $fIle_url is is reachable because of parent function

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 9]
```diff
@@ -365,13 +397,13 @@ function process_zip_list_file {
 
 		check_if_tag_exists liferay-dxp "${tag_name_new}" && continue
 
-		check_ignore_zip_file "${hotfix_zip_file}" "${release_version}" && continue
+		check_ignore_zip_file "${hotfix_zip_file}" && continue
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh:400]: Why did you remove $release_version?

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh [hunk 10]
```diff
@@ -365,13 +397,13 @@ function process_zip_list_file {
...
 
-		file_url="${zip_directory_url}/${hotfix_zip_file}"
+		zip_file_path="${zip_directory_url}/${hotfix_zip_file}"
```
Comment (tomposmiko) [inline narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: Why did you change the variable name?

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh (hunk 9 above)
Comment (Wykenz) [reply narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh:400]: It was before we talked about putting there arguments that the code don't need. Should I put it back?

File: narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh (hunk 10 above)
Comment (Wykenz) [reply narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh]: You said that file_url is not the best for it and it should be something with "path"
